packages/app/src/vis-packs/core/heatmap/MappedHeatmapVis.tsx

```diff
@@ -58,22 +65,39 @@ function MappedHeatmapVis(props: Props) {
     customDomain,
     colorMap,
     scaleType,
+    complexVisType,
     keepRatio,
     showGrid,
     invertColorMap,
     flipXAxis,
     flipYAxis,
   } = config;
 
-  const numArray = useToNumArray(value);
+  const { shape: dims, type } = dataset;
+  const isComplex = isComplexType(type);
+
+  // If complex dataset displayed as phase/amplitude: `numArray` = amplitude; `alphaArray` = phase
+  const isPhaseAmp = complexVisType === ComplexVisType.PhaseAmplitude;
+
+  const numArray = useToNumArray(
+    value,
+    isPhaseAmp ? ComplexVisType.Amplitude : complexVisType,
+  );
+  const phaseArray = useToNumArray(
+    isComplex && isPhaseAmp ? value : undefined,
+    ComplexVisType.Phase,
+  );
+
```

Comment:
This is the slightly complicated logic to account for the combined phase/amplitude case.